ls1intum/Artemis
# `Development`: Refactor course registration

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This is a follow-up PR to https://github.com/ls1intum/Artemis/pull/4353, in which the course registration was moved into its own page in order to improve the initial bundle size. Some good suggestions were made on how to improve the code quality of this component and they are addressed here.

### Description

Layout and styling of the courses list is improved. Error and success messages are now displayed using the alert service. The Courses, for which a student can register, are now fetched directly in one API call instead of two.

### Steps for Testing

Prerequisites:
- 1 Students
- At least one course for which the stundet can register

1. Log in to Artemis
2. Navigate to Course Registration
3. Check that the correct registrable courses are displayed
4. Register for the course
5. Check that the registration still works as expected and the student is registered for the course (e.g. it's displayed on the landing page)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





course-registration.component.ts: 52% | 94% (relatively low branch coverage is due to one generic error handling case not being covered)

### Screenshots


<img width="1437" alt="Bildschirmfoto 2021-12-20 um 15 23 00" src="https://user-images.githubusercontent.com/51987021/146815778-f717e1ea-c2d5-407d-b940-a63d6e26e3d3.png">
